crates/apub/src/api/list_posts.rs

```diff
@@ -88,6 +91,14 @@ pub async fn list_posts(
   .await
   .with_lemmy_type(LemmyErrorType::CouldntGetPosts)?;
 
+  // If in their user settings, auto-mark fetched posts as read
+  if let Some(local_user) = local_user {
+    if local_user.auto_mark_fetched_posts_as_read {
+      let post_ids = posts.iter().map(|p| p.post.id).collect::<Vec<PostId>>();
+      PostRead::mark_as_read(&mut context.pool(), &post_ids, local_user.person_id).await?;
+    }
+  }
```

Comment:
Main important change is here. 

Not really much need to test it since post_view already has tests for read posts.